modules/core/src/main/java/org/apache/ignite/internal/processors/cache/DynamicCacheDescriptor.java

```diff
@@ -387,7 +409,15 @@ public QuerySchemaPatch makeSchemaPatch(Collection<QueryEntity> target) {
      */
     public boolean applySchemaPatch(QuerySchemaPatch patch) {
         synchronized (schemaMux) {
-            return schema.applyPatch(patch);
+            boolean res = schema.applyPatch(patch);
+
+            Optional<SchemaAbstractOperation> op = patch.getPatchOperations().stream()
```

Comment:
Let's use filter+forEach here instead of findFirst and isPresent.

```

            if (res) {
                patch.getPatchOperations().stream()
                    .filter(op -> op instanceof SchemaAddQueryEntityOperation)
                    .forEach(op -> cacheCfg = patchCacheConfiguration(cacheCfg, (SchemaAddQueryEntityOperation)op));
            }
```